samples/AlarmBot/Startup.cs

```diff
@@ -42,11 +43,18 @@ public void ConfigureServices(IServiceCollection services)
                     {
                         await context.SendActivity("Sorry, it looks like something went wrong!");
                     }));
+
+                // Add and configure state management middleware
+                middleware.Add(new StateManagementMiddleware()
+                                .UseDefaultStorageProvider(new MemoryStateStorageProvider())

```

Comment:
>UseDefaultStorageProvider [](start = 33, length = 25)

mmm.seems like SetDefaultStorageProvider() followed by .Use() 
don't feel super strongly about this one i guess

Reply:
Yeah I do think the name is confusing as well. It's just that we use the `UseXXX` verb almost everywhere and `UseDefaultStorageProvider` does ultimately do what `UseStorageProvider` does. In fact it literally defers to the `UseStorageProvider` implementation under the covers, it just passes the well-known key that the system looks up the default provider by.

I think the confusing thing about the name though is that it makes it sound like you're asking to use "the" default storage provider which is not what we're saying here.

I thought about maybe `UseStorageProviderAsDefault`? Dunnuh if I love that either, but it's technically more correct.